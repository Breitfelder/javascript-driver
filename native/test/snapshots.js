import fs from 'fs';
import { guessParsing as parse } from '../lib';
import test from 'ava';

function isDir(path) {
  return fs.statSync(path).isDirectory();
}

function* getAllFiles(path) {
  let files = fs.readdirSync(path);
  for(let filename of files) {
    let completePath = `${path}/${filename}`;
    if (isDir(completePath)) {
      yield *getAllFiles(completePath);
    } else {
      yield completePath;
    }
  }
}

// TODO: Improve once this is working
function snapshotDir(t, path, cb) {
  for(let filepath of getAllFiles(path)) {
    if (!filepath.endsWith('.gitkeep')) {
      fs.readFile(filepath, 'utf8', function(err, content) {
        t.ifError(err, `could not read ${filepath}`);

        if (!err) {
          cb(filepath, content);
        }
      });
    }
  }
}

// Skip until avajs/ava#1223 gets solved. For now, snapshots are unreliable for
// our use case.
test.skip(`snapshot fixtures`, t => {
  snapshotDir(t, `${__dirname}/fixtures`, function(filepath, content) {
    let ast = parse(content);
    t.snapshot(ast, `snapshot for ${filepath}`)
    t.truthy(ast, `${filepath} produced an ast`);
  });
});
