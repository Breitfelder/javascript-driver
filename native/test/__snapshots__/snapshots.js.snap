exports[`snapshot /Users/serabe/programming/babelfish-js-driver/test/fixtures/parse-strict/sample-001.js 1`] = `"{\"type\":\"File\",\"start\":0,\"end\":68,\"loc\":{\"start\":{\"line\":1,\"column\":0},\"end\":{\"line\":4,\"column\":0}},\"program\":{\"type\":\"Program\",\"start\":0,\"end\":68,\"loc\":{\"start\":{\"line\":1,\"column\":0},\"end\":{\"line\":4,\"column\":0}},\"sourceType\":\"module\",\"body\":[{\"type\":\"ImportDeclaration\",\"start\":0,\"end\":34,\"loc\":{\"start\":{\"line\":1,\"column\":0},\"end\":{\"line\":1,\"column\":34}},\"specifiers\":[{\"type\":\"ImportDefaultSpecifier\",\"start\":7,\"end\":16,\"loc\":{\"start\":{\"line\":1,\"column\":7},\"end\":{\"line\":1,\"column\":16}},\"local\":{\"type\":\"Identifier\",\"start\":7,\"end\":16,\"loc\":{\"start\":{\"line\":1,\"column\":7},\"end\":{\"line\":1,\"column\":16},\"identifierName\":\"something\"},\"name\":\"something\"}}],\"importKind\":\"value\",\"source\":{\"type\":\"StringLiteral\",\"start\":22,\"end\":33,\"loc\":{\"start\":{\"line\":1,\"column\":22},\"end\":{\"line\":1,\"column\":33}},\"extra\":{\"rawValue\":\"somewhere\",\"raw\":\"\'somewhere\'\"},\"value\":\"somewhere\"}},{\"type\":\"ExportDefaultDeclaration\",\"start\":36,\"end\":67,\"loc\":{\"start\":{\"line\":3,\"column\":0},\"end\":{\"line\":3,\"column\":31}},\"declaration\":{\"type\":\"BinaryExpression\",\"start\":51,\"end\":66,\"loc\":{\"start\":{\"line\":3,\"column\":15},\"end\":{\"line\":3,\"column\":30}},\"left\":{\"type\":\"Identifier\",\"start\":51,\"end\":60,\"loc\":{\"start\":{\"line\":3,\"column\":15},\"end\":{\"line\":3,\"column\":24},\"identifierName\":\"something\"},\"name\":\"something\"},\"operator\":\"!=\",\"right\":{\"type\":\"NumericLiteral\",\"start\":64,\"end\":66,\"loc\":{\"start\":{\"line\":3,\"column\":28},\"end\":{\"line\":3,\"column\":30}},\"extra\":{\"rawValue\":42,\"raw\":\"42\"},\"value\":42}}}],\"directives\":[]},\"comments\":[],\"tokens\":[{\"type\":{\"label\":\"import\",\"keyword\":\"import\",\"beforeExpr\":false,\"startsExpr\":false,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"value\":\"import\",\"start\":0,\"end\":6,\"loc\":{\"start\":{\"line\":1,\"column\":0},\"end\":{\"line\":1,\"column\":6}}},{\"type\":{\"label\":\"name\",\"beforeExpr\":false,\"startsExpr\":true,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null},\"value\":\"something\",\"start\":7,\"end\":16,\"loc\":{\"start\":{\"line\":1,\"column\":7},\"end\":{\"line\":1,\"column\":16}}},{\"type\":{\"label\":\"name\",\"beforeExpr\":false,\"startsExpr\":true,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null},\"value\":\"from\",\"start\":17,\"end\":21,\"loc\":{\"start\":{\"line\":1,\"column\":17},\"end\":{\"line\":1,\"column\":21}}},{\"type\":{\"label\":\"string\",\"beforeExpr\":false,\"startsExpr\":true,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"value\":\"somewhere\",\"start\":22,\"end\":33,\"loc\":{\"start\":{\"line\":1,\"column\":22},\"end\":{\"line\":1,\"column\":33}}},{\"type\":{\"label\":\";\",\"beforeExpr\":true,\"startsExpr\":false,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"start\":33,\"end\":34,\"loc\":{\"start\":{\"line\":1,\"column\":33},\"end\":{\"line\":1,\"column\":34}}},{\"type\":{\"label\":\"export\",\"keyword\":\"export\",\"beforeExpr\":false,\"startsExpr\":false,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"value\":\"export\",\"start\":36,\"end\":42,\"loc\":{\"start\":{\"line\":3,\"column\":0},\"end\":{\"line\":3,\"column\":6}}},{\"type\":{\"label\":\"default\",\"keyword\":\"default\",\"beforeExpr\":true,\"startsExpr\":false,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"value\":\"default\",\"start\":43,\"end\":50,\"loc\":{\"start\":{\"line\":3,\"column\":7},\"end\":{\"line\":3,\"column\":14}}},{\"type\":{\"label\":\"name\",\"beforeExpr\":false,\"startsExpr\":true,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null},\"value\":\"something\",\"start\":51,\"end\":60,\"loc\":{\"start\":{\"line\":3,\"column\":15},\"end\":{\"line\":3,\"column\":24}}},{\"type\":{\"label\":\"==/!=\",\"beforeExpr\":true,\"startsExpr\":false,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":6,\"updateContext\":null},\"value\":\"!=\",\"start\":61,\"end\":63,\"loc\":{\"start\":{\"line\":3,\"column\":25},\"end\":{\"line\":3,\"column\":27}}},{\"type\":{\"label\":\"num\",\"beforeExpr\":false,\"startsExpr\":true,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"value\":42,\"start\":64,\"end\":66,\"loc\":{\"start\":{\"line\":3,\"column\":28},\"end\":{\"line\":3,\"column\":30}}},{\"type\":{\"label\":\";\",\"beforeExpr\":true,\"startsExpr\":false,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"start\":66,\"end\":67,\"loc\":{\"start\":{\"line\":3,\"column\":30},\"end\":{\"line\":3,\"column\":31}}},{\"type\":{\"label\":\"eof\",\"beforeExpr\":false,\"startsExpr\":false,\"rightAssociative\":false,\"isLoop\":false,\"isAssign\":false,\"prefix\":false,\"postfix\":false,\"binop\":null,\"updateContext\":null},\"start\":68,\"end\":68,\"loc\":{\"start\":{\"line\":4,\"column\":0},\"end\":{\"line\":4,\"column\":0}}}]}"`;
